aea/test_tools/generic.py

```diff
@@ -117,13 +117,10 @@ def get_nested_ordered_dict_from_keys_and_value(
         return _dic
 
     root_key = keys[0]
-    if (
-        isinstance(configuration_obj, SkillConfig)
-        and root_key in SkillConfig.FIELDS_WITH_NESTED_FIELDS
-    ):
+    if isinstance(configuration_obj, SkillConfig):
```

Comment:
why limited to skill config?

Reply:
shouldn't you keep the check for root_key in FIELDS_WITH_NESTED_FIELDS and do it for all configs?